### PR TITLE
isup plugin: checks if a website is up or not.

### PR DESCRIPTION
**Pull request by Cameri**
This plugin verifies if a website is up or not.

### CODE REVIEW

**Comment by yagop** [comment]
I think this should be done on [ping](https://github.com/yagop/telegram-bot/blob/master/plugins/ping.lua) plugin. And I also think is better to check only the socket connexion much faster and easier. Maintain the `!command` convention.

**Comment by Cameri** [comment]
I thought I had changed it back to the exclamation mark. My bad. 
Do you consider a website returning status code 500 up or down? If you only check if the socket opens successfully, I'm sure you won't be able to tell.

Cheers.

**Comment by Cameri** [comment]
See here: https://github.com/al3x/downforeveryoneorjustme/blob/master/main.py
Also here: https://github.com/MikeRixWolfe/Gary/blob/master/plugins/isup.py

**Comment by yagop** [comment]
> Do you consider a website returning status code 500 up or down?

I consider it up, but you are right to check because user doesn't know if it's his own session/browser/whatever what is failing.  However I would return  error if `>= 500`, not `404` or warns the user.

There are also reasons to check the HTTP connexion like reverse proxies and Virtual hosts used by the servers.

The matter I tried to solve with socket checking was the delay of the response, a HTTP connexion takes much more time because of the response data. A website with too much HTML content may delay too long. I would use a timeout (by default I use 10s) on the HTTP connexion in order to request other users (telegram-bot is synchronous) commands. What if the timeout expires and the website keep sending data? There is no way to control that state. Obviously is not the common one but I think I was too paranoid when I wrote that plugin that forgot other more common scenarios (written before).

> See here: https://github.com/al3x/downforeveryoneorjustme/blob/master/main.py
> Also here: https://github.com/MikeRixWolfe/Gary/blob/master/plugins/isup.py

To be honest, I don't like those implementations, they aren't hackish.

In summary: You win, there are many plugins, so only one file will be keep (`isup.lua`) but I will add the cron functionality to it. And a way to test the socket when a specific port is provided. Eg `!isup 8.8.8.8:53`
